firedrake/petsc.py

```diff
@@ -188,10 +188,18 @@ def set_from_options(self, petsc_obj):
     def inserted_options(self):
         """Context manager inside which the petsc options database
     contains the parameters from this object."""
+        from firedrake.logging import warning
         try:
             for k, v in self.parameters.items():
                 key = self.options_prefix + k
                 if type(v) is bool:
```

Comment:
Once the deprecation period is up, we just remove the condition and this turns into:
```
for k, v in self.parameters.items():
    self.options_object[self.options_prefix + k] = v
```